provider/management/commands/delete_expired_grant_tokens.py

```diff
@@ -0,0 +1,53 @@
+"""
+Command to delete expired OAuth2 grant tokens from the oauth2_grant table.
+"""
+
+import logging
+from datetime import datetime, timedelta
+from django.core.management.base import BaseCommand
+from django.db import transaction
+from provider.oauth2.models import Grant
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage: ./manage.py lms --settings=devstack delete_expired_grant_tokens.py
+    """
+    help = 'Deletes all expired OAuth2 grant tokens (in chunks).'
+
+    # Maximum number of expired tokens to delete in a single transaction.
+    CHUNK_SIZE = 10000
+
+    def handle(self, *args, **options):
+        """
+        Deletes oauth2_grant tokens expired for days_back days, chunking the deletes to avoid long table/row locks.
+        """
+        table_name = 'oauth2_grant'
+        delete_date = datetime.now()
+        where_clause = "WHERE expires < '{}'".format(delete_date.strftime('%Y-%m-%d %H:%m:%S'))
+        # The "as id" below fools Django raw query into thinking the primary key is being queried.
+        total_expired_tokens = Grant.objects.raw(
+            'SELECT COUNT(*) as id FROM {} {}'.format(table_name, where_clause)
+        )[0].id
+        log.info("STARTED: Deleting %s expired grant tokens older than %s.", total_expired_tokens, delete_date)
+
+        total_deletions = 0
+        while total_deletions < total_expired_tokens:
+            deletions_now = min(self.CHUNK_SIZE, total_expired_tokens - total_deletions)
+            log.info("Deleting {} expired tokens...".format(deletions_now))
+            with transaction.atomic():
+                delete_sql = 'DELETE FROM {} WHERE id IN (SELECT id FROM {} {} LIMIT {})'.format(
+                    table_name, table_name, where_clause, self.CHUNK_SIZE
```

Comment:
Did you mean to use `deletions_now` instead of `self.CHUNK_SIZE` here? Otherwise the log statement and `total_deletions` could be wrong.